benchmarks/_benchmark.py

```diff
@@ -86,10 +141,27 @@ def benchmark(
                 context=context,
                 github=self.github_info,
                 options=options,
+                publish=os.environ.get("DRY_RUN") is None,
+            )
+
+            res = BenchmarkResult(
+                name=benchmark["tags"]["name"],
+                run_id=benchmark.get("run_id"),
+                batch_id=benchmark.get("batch_id"),
+                result=benchmark.get("stats"),
+                params=params,
+                tags=benchmark.get("tags"),
+                info=benchmark.get("info"),
+                context=benchmark.get("context"),
+                github=benchmark.get("github"),
+                options=benchmark.get("options"),
+                output=output,
+                error=benchmark.get("error"),
             )
         except Exception as e:
-            benchmark, output = self._handle_error(e, self.name, tags, info, context)
-        return benchmark, output
+            res = self._handle_error(e, self.name, tags, info, context)
+
+        return res
```

Comment:
This will not work. Conbench runner needs `benchmark, output` to be returned by benchmark method.